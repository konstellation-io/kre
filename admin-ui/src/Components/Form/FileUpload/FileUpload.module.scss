@import 'Styles/mixins';
@import 'Styles/colors';

.container {
  position: relative;

  &.white,
  &.white .input {
    color: white;
  }

  &.hasClearButton .input {
    padding-right: 30px;
  }
}

.inputContainer {
  display: flex;
  flex-direction: row;
}

.input {
  @include form-input;

  color: font-color(light);
  font-weight: 600;

  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  &.placeholder {
    color: font-color(regular);
  }
}
