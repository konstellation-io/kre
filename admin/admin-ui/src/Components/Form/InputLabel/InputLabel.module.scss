@import 'Styles/mixins';
@import 'Styles/colors';

.label {
  @include font-caption;

  display: block;
  letter-spacing: 0.4px;
  color: font-color(grey);
  margin-bottom: 11px;
  min-height: 13px;

  transition: color ease 0.3s;

  &.hidden {
    color: transparent;
  }
}
