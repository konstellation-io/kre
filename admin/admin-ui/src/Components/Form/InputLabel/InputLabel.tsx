import React from 'react';
import cx from 'classnames';
import styles from './InputLabel.module.scss';

type Props = {
  text?: string;
  hidden?: boolean;
};

function InputLabel({ text = '', hidden = false }: Props) {
  return (
    <label
      className={cx(styles.label, { [styles.hidden]: hidden })}
      data-testid="label"
    >
      {text.toUpperCase()}
    </label>
  );
}

export default InputLabel;
