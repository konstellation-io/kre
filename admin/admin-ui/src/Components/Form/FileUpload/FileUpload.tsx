import React, { useRef, useState } from 'react';

import Button from '../../Button/Button';
import InputError from '../InputError/InputError';
import InputLabel from '../InputLabel/InputLabel';
import cx from 'classnames';
import { get } from 'lodash';
import styles from './FileUpload.module.scss';

type Props = {
  onChange?: Function;
  name?: string;
  placeholder?: string;
  label?: string;
  height?: number;
  error?: string;
  inputRef?: React.Ref<any>;
  autofocus?: boolean;
};

function FileUpload({
  onChange = function() {},
  name = 'file-upload',
  placeholder = '',
  label = '',
  height = 40,
  error = '',
  inputRef = null,
  autofocus = false
}: Props) {
  const [selectedFile, setSelectedFile] = useState<File>();
  const fileButton = useRef<HTMLInputElement>(null);

  function onFileUpload() {
    const fileButtonEl =
      fileButton.current && fileButton.current.querySelector('input');
    if (fileButtonEl !== null && fileButtonEl.files !== null) {
      const file = fileButtonEl.files[0];
      setSelectedFile(file);
      onChange(file);
    }
  }

  const inputText = get(selectedFile, 'name', placeholder);

  return (
    <div ref={fileButton} className={styles.container}>
      <InputLabel text={label} />
      <div className={styles.inputContainer}>
        <div
          className={cx(styles.input, {
            [styles.placeholder]: !selectedFile,
            [styles.error]: error !== ''
          })}
          style={{ height }}
          data-testid="input"
          title={inputText}
        >
          {inputText}
        </div>
        <Button
          label="BROWSE"
          onClick={() => {
            const submitButton =
              fileButton.current && fileButton.current.querySelector('input');
            if (submitButton !== null) {
              submitButton.click();
            }
          }}
          tabIndex={0}
          border
          autofocus={autofocus}
        />
      </div>
      <input
        name={name}
        type="file"
        ref={inputRef}
        onChange={onFileUpload}
        hidden
      />
      <InputError message={error} />
    </div>
  );
}

export default FileUpload;
