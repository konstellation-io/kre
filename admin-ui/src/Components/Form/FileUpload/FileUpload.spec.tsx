import React from 'react';
import FileUpload from './FileUpload';
import '@testing-library/jest-dom/extend-expect';
import { shallow } from 'enzyme';
import InputLabel from '../InputLabel/InputLabel';

describe('FileUpload', () => {
  let wrapper: any;
  const PLACEHOLDER = 'some placeholder';
  const LABEL = 'some label';

  beforeEach(() => {
    wrapper = shallow(<FileUpload placeholder={PLACEHOLDER} label={LABEL} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('show right texts', () => {
    expect(wrapper.find(InputLabel).prop('text')).toBe(LABEL);
    expect(wrapper.find('.input').text()).toBe(PLACEHOLDER);
  });
});
