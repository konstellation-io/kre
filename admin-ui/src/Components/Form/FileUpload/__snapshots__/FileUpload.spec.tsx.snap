// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload matches snapshot 1`] = `
<div
  className="container"
>
  <InputLabel
    text="some label"
  />
  <div
    className="inputContainer"
  >
    <div
      className="input placeholder"
      data-testid="input"
      style={
        Object {
          "height": 40,
        }
      }
      title="some placeholder"
    >
      some placeholder
    </div>
    <Button
      border={true}
      label="BROWSE"
      onClick={[Function]}
    />
  </div>
  <input
    hidden={true}
    name="file-upload"
    onChange={[Function]}
    type="file"
  />
  <InputError
    message=""
  />
</div>
`;
